### ISSUE TITLE
Project "hellocardboard-android" fails to build in Android Studio Giraffe

### ISSUE THREAD

**Issue by lneumarkt** [googlevr/cardboard#439]
### SPECIFIC ISSUE ENCOUNTERED

When following the [Quickstart](https://developers.google.com/cardboard/develop/c/quickstart) tutorial to build the project `hellocardboard-android`, the following error appears on Android Studio Giraffe:

```
Unable to make field private final java.lang.String java.io.File.path accessible: module java.base does not "opens java.io" to unnamed module @40c8ebe1
```

#### Workaround
I was able to solve this issue by adding the following line to `gradle.properties`:
```
org.gradle.jvmargs=-Xmx1536M --add-exports=java.base/sun.nio.ch=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED
```
With these changes the project should build and run.

### SMARTPHONE:
- Device: Samsung S20 FE
- OS
  - [x] Android
  - [ ] iOS
- OS version 12

### STEPS TO REPRODUCE THE ISSUE
1. Follow the [Download and build the demo app ](https://developers.google.com/cardboard/develop/c/quickstart#download_and_build_the_demo_app) instructions from the [Quickstart](https://developers.google.com/cardboard/develop/c/quickstart) tutorial.
2. The error appears when attempting step `4. Run the HelloCardboard demo app`.

### EXPECTED BEHAVIOR
The app should be built without issues and deployed successfully to the smartphone.

### VERSIONS USED
Android Studio Giraffe | 2022.3.1 Patch 2

**Comment by arilow** [comment]
Thank you for reporting this issue. I confirm I’m able to repro de issue locally. It has been added to our backlog and will be prioritized along with other issues. 

We'll update this issue once we have some updates about it.

**Comment by lneumarkt** [comment]
Thanks @Mandelbrow for reaching out to us!

We were able to reproduce the reported behavior. However, that is intended. 

The short URI goes through a number of redirections in [CardboardParamsUtils.java](https://github.com/googlevr/cardboard/blob/5c9f3066dc14962d1dec9a32ec9d3668641c408d/sdk/qrcode/android/java/com/google/cardboard/sdk/qrcode/CardboardParamsUtils.java#L177) to expand to the long URI, and one of those redirection matches the intended pattern [cardboard/cfg](https://github.com/googlevr/cardboard/blob/5c9f3066dc14962d1dec9a32ec9d3668641c408d/sdk/qrcode/android/java/com/google/cardboard/sdk/qrcode/CardboardParamsUtils.java#L66) before reaching the last redirection, which corresponds to https://arvr.google.com/cardboard/download/?p=deviceParams 

In order to pass the long URI it needs to be in the format described in the [Cardboard documentation](https://developers.google.com/cardboard/reference/unity/class/Google/XR/Cardboard/Api#savedeviceparams). 

For​​ `http://goo.gl/Rw4kwC` , the URI that Cardboard uses to get the viewer parameters is :
https://google.com/cardboard/cfg?p=ChlVbm9mZmljaWFsIENhcmRib2FyZCBJbmMuEgtUaGUgQ2xhc3NpYx3sUTg9JY_CdT0qEAAASEIAAEhCAABIQgAASEJYADVQjRc9OggAAAAAAAAAAFABYAE 

Please feel free to reopen this ticket in case you have any other questions regarding this topic. Thanks!

**Comment by jballoffet** [comment]
A fix for this issue has been submitted and it will be included in v1.23. Thanks for reaching out!

**Comment by xinyunh0929** [comment]
This was fixed in v1.23.0
